### PR TITLE
Added type to options

### PR DESCRIPTION
**Pull request by raimonomiar**
Added type to the cli options since some of them were missing and made minor code changes.
Moved port to config file.
Added npm start script for example: `npm start -- --help`
Fixed the undefined issue https://github.com/parvardegr/sharing/issues/31

### CODE REVIEW

File: bin/index.js [hunk 1]
```diff
@@ -63,31 +63,27 @@ var debugLog = (log) => {
...
-        .option("debug", { describe: "enable debuging logs", demandOption: false })
-        .option("p", { alias: 'port', describe: "Change default port", demandOption: false })
-        .option("ip", { describe: "Your machine public ip address", demandOption: false })
-        .option("c", { alias: 'clipboard', describe: "Share Clipboard", demandOption: false })
-        .option("w", { alias: 'on-windows-native-terminal', describe: "Enable QR-Code support for windows native terminal", demandOption: false })
-        .option("r", { alias: 'receive', describe: "Receive files", demandOption: false })
-        .option("q", { alias: 'receive-port', describe: "change receive default port", demandOption: false })
-        .option("U", { default: 'user', alias: 'username', describe: "set basic authentication username", demandOption: false })
-        .option("P", { alias: 'password', describe: "set basic authentication password", demandOption: false })
+        .option("debug", { describe: "enable debuging logs", demandOption: false, type: 'boolean' })
```
Comment (parvardegr) [inline bin/index.js]: i did’t add type for the options intentionally, there is no need for them, they just make —help output noisy. can you revert this parts?

other changes are nice 🙏

File: bin/config.js [hunk 2]
```diff
@@ -13,5 +13,9 @@ module.exports = {
         protocolModule: http,
         protocol: 'http',
         option: {}
-    }
+    },
+    appPort: 7478,
```
Comment (parvardegr) [inline bin/config.js]: if you add default prefix to these new values it would be so nice, ex: defaultAppPort: 7478

File: bin/index.js [hunk 3]
```diff
@@ -162,7 +158,7 @@ var debugLog = (log) => {
     
     if (options.receive) {
         const app = createDefaultApp();
-        let uploadAddress = options.ip? `${config.ssl.protocol}://${options.ip}:${options.receivePort}/receive`: `${config.ssl.protocol}://${getNetworkAddress()}:${options.receivePort}/receive`;
+        let uploadAddress = options.ip && options.receivePort ? `${config.ssl.protocol}://${options.ip}:${options.receivePort}/receive`: `${config.ssl.protocol}://${getNetworkAddress()}:${config.receivePort}/receive`;
         console.log(uploadAddress);
```
Comment (parvardegr) [inline bin/index.js]: remove this console.log and you don't need to change this line.
it’s here by mistake

File: bin/index.js (hunk 1 above)
Comment (raimonomiar) [reply bin/index.js]: Removed all the type from cli options.

File: bin/config.js (hunk 2 above)
Comment (raimonomiar) [reply bin/config.js]: Added prefix.

File: bin/index.js (hunk 3 above)
Comment (raimonomiar) [reply bin/index.js]: removed console.log

**Comment by parvardegr** [comment]
@raimonomiar very nice 🙏

File: bin/index.js [hunk 4]
```diff
@@ -209,11 +203,11 @@ var debugLog = (log) => {
             startServer(app, options.receivePort, listener);
         else {
             portfinder.getPort({
-                port: 1374,
-                stopPort: 1400
+                port: config.defaultReceivePort,
+                stopPort: config.defaultReceivePort
```
Comment (parvardegr) [inline bin/index.js]: a typo here

Comment (raimonomiar) [reply bin/index.js]: Updated

**Pull request merged**